crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -493,13 +493,15 @@ where
 /// capabilities and the input list of locally supported capabilities.
 ///
 /// Currently only `eth` versions 66 and 67 are supported.
+// ANDREW: We should also mention that the `p2p` protocol (version 5?) is supported,
+// but it's not expected to be in `local_capabilities`
 pub fn set_capability_offsets(
     local_capabilities: Vec<Capability>,
     peer_capabilities: Vec<Capability>,
+    // ANDREW: Why does this only return a single shared capability?
```

Comment:
Pls leave questions as PR comments instead of inline in the code.

@Rjected thoughts?

Reply:
This only returns a single capability mainly because the current stream design would require a router in order to support multiple. In other words, the `EthStream` directly wraps the `P2PStream`, assumes `eth` is the only supported capability and just subtracts the offset for every message in the stream before decoding.

To support multiple capabilities we'd need to create a type (which might need to implement `Future` or something similar) which awaits on the `P2PStream`, determines the intended capability (based on the shared capabilities / offsets), and sends the message to that capabilities' stream.

This is a much larger feature / refactor but would be required if we were to support more than one capability (`snap` and `eth` for example)

does that make sense?

Reply:
Got it. So if we were to support snap/les or multiple capabilities we'd need to do a bigger refactor. I think fine for now, but let's track it somewhere that this is a prerequisite.

Reply:
tracking this in #791 , will add the explanation above for context